src/modules/authentication/AuthenticationWrapper.ts

```diff
@@ -113,13 +129,28 @@ export class AuthenticationWrapper implements IAuthenticationWrapper {
       return result;
     } catch (error) {
       if (error instanceof InteractionRequiredAuthError || !this.getAccount()) {
+
+        return this.loginWithInteraction(silentRequest.scopes, sessionId);
+
+      } else if(this.checkIfAuthResultError(error)) {
+        this.deleteHomeAccountId();
         return this.loginWithInteraction(silentRequest.scopes, sessionId);
-      } else {
+      }
+      else{
         throw error;
       }
     }
   }
 
+  private checkIfAuthResultError(errorCode: string): boolean {
+    for(const errorItem of AuthErrorList){
+      if( errorItem.trim() === errorCode.trim() ){
+        return true;
+      }
+    }
+    return false;
```

Comment:
This could give the same result

```suggestion
    return AuthErrorList.includes(errorCode.trim());
```

Reply:
Yeah. I've seen it. Shorter and cleaner. Let me fix this also